lib/generators/plugin/migration_generator.rb

```diff
@@ -0,0 +1,72 @@
+module Plugin
+  class MigrationGenerator < Rails::Generators::Base
+    class_option :plugin_name, :required => true
+    class_option :plugin_source
+
+    attr_accessor :migration_pattern
+
+    hook_for :orm, :required => true do |instance, klass|
+      new_migration = instance.invoke klass, nil, nil, destination_root: instance.send(:migration_destination)
+      # the task return a suggested new migration name. If the migration already
+      # exists, the file with older timestamp would remain. In this case we
+      # should search for the older file with the same name part.
+      instance.migration_pattern = new_migration.first
+    end
+
+    def initialize(*args)
+      super
+      self.destination_root = migration_destination
+    end
+
+    def copy_latest_migration
+      destination_file = File.join(
+        destination_root,
+        'db/migrate',
+        destination_file_name
+      )
+
+      unless options['pretend']
+        # There is no rename task for some reason...
+        copy_file migration_file, destination_file
+        remove_file File.join('db/migrate', migration_file)
+      end
+      [destination_file]
+    end
+
+    private
+
+    def extract_migration_name(migration_file_name)
+      File.basename(migration_file_name).sub(/^\d*_/, '').sub(/\.rb$/, '')
+    end
+
+    def source_paths
+      [File.join(migration_destination, 'db/migrate')]
+    end
+
+    def migration_destination
+      File.join(destination_root, plugin_source)
+    end
+
+    def destination_file_name
+      File.basename(migration_file, File.extname(migration_file)) +
+        ".#{plugin_name}" +
+        File.extname(migration_file)
+    end
+
+    def migration_file
+      @migration_file ||= begin
+        full_name = Dir.glob(File.join(migration_destination + 'db/migrate/*.rb')).grep(/\d+_#{migration_pattern}.rb$/).first
+        File.basename(full_name)
```

Comment:
getting ugly trace on this line

```
/home/ares/Projekty/Zdrojaky/foreman/lib/generators/plugin/migration_generator.rb:62:in `basename': no implicit conversion of nil into String (TypeError)
        from /home/ares/Projekty/Zdrojaky/foreman/lib/generators/plugin/migration_generator.rb:62:in `migration_file'
        from /home/ares/Projekty/Zdrojaky/foreman/lib/generators/plugin/migration_generator.rb:51:in `destination_file_name'
        from /home/ares/Projekty/Zdrojaky/foreman/lib/generators/plugin/migration_generator.rb:25:in `copy_latest_migration'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/invocation.rb:133:in `block in invoke_all'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/invocation.rb:133:in `each'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/invocation.rb:133:in `map'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/invocation.rb:133:in `invoke_all'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/group.rb:232:in `dispatch'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/thor-0.19.4/lib/thor/base.rb:444:in `start'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.8/lib/rails/generators.rb:157:in `invoke'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.8/lib/rails/commands/generate.rb:13:in `<top (required)>'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `block in require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.8/lib/rails/commands/commands_tasks.rb:123:in `require_command!'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.8/lib/rails/commands/commands_tasks.rb:130:in `generate_or_destroy'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.8/lib/rails/commands/commands_tasks.rb:50:in `generate'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.8/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/railties-4.2.8/lib/rails/commands.rb:17:in `<top (required)>'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `block in require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:274:in `require'
        from /home/ares/Projekty/Zdrojaky/foreman/bin/rails:9:in `<top (required)>'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `block in load'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
```

the path it constructs is wrong - `/home/ares/Projekty/Zdrojaky/foreman_openscap/../foreman_openscap/db/migrate/*.rb`, note that the foreman dir from which I run it is at `/home/ares/Projekty/Zdrojaky/foreman`. It seems that `migration_destination` returns wrong directory.